server/routes/api/protein.py

```diff
@@ -25,10 +30,300 @@
+    '''
+    scores_sym = {}
+    for interaction_id, score in scores.items():
+        reverse_id = reverse_interaction_id(interaction_id)
+        reverse_score = scores.get(reverse_id, DEFAULT_INTERACTION_SCORE)
+        scores_sym[interaction_id] = scores_sym[reverse_id] = max(
+            score, reverse_score)
+    return scores_sym
+
+
+def flatten(nested_list):
+    '''
+    Given a list of lists, return a depth-1 flattened version
+    '''
+    flat = []
+    for L in nested_list:
+        flat.extend(L)
+    return flat
+
+
+def partition_expansion_cross(target_ids, node_set):
+    '''
+    Return two lists expansion_targets, cross_targets such that
+    1) expansion_targets U cross_targets = target_ids
+    2) expansion_targets and cross_targets are disjoint
+    3) all elements of cross_targets are in node_set
+    4) (implied) no elements of expansion_targets are in node_set
+    5) Stability: If id1 appears before id2 in target_ids and they are both assigned to the same
+       partition (either expansion_targets or cross_targets), id1 appears before id2 in this partition too.
+    '''
+    expansion_targets = []
+    cross_targets = []
+    for target_id in target_ids:
+        if target_id in node_set:
+            cross_targets.append(target_id)
+        else:
+            expansion_targets.append(target_id)
+    return expansion_targets, cross_targets
+
+
+@bp.route('/ppi/bfs/', methods=['POST'])
+def bfs():
+    '''
+    Retrieves graph data for protein-protein interaction graph via degree-bounded BFS
+
+    Request params:
+    - proteinDcid: the DCID of the page protein (e.g. 'bio/P53_HUMAN')
+    - scoreThreshold: interaction score threshold above which to display an edge between two interacting proteins
+    - maxInteractors: we take {numInteractors} top-scoring expansion links of each node during BFS
+    - depth: the maximum distance of any returned node from the center protein
+
+    Returns: A dict of form
+    {
+        'nodeDataNested': [[...depth 0 node dicts], [...depth 1 node dicts], ...]
+        'linkDataNested': [[...depth 0 link dicts], [...depth 1 link dicts], ...]
+    }
+    The only depth 0 node is the center protein and since we do not support self-interactions at the moment,
+    there are no depth-0 links, so the above can be simplified to
+    {
+        'nodeDataNested': [[center protein dict], [...center protein neighbor dicts], ...]
+        'linkDataNested': [[], [...center protein link dicts], ...]
+    }.
+
+    See server/tests/test_data/protein/P53_HUMAN_small.json for an example response for request parameters
+    {
+        'depth': 2,
+        'maxInteractors': 2,
+        'proteinDcid': 'bio/P53_HUMAN',
+        'scoreThreshold': 0.4
+    }.
+    '''
+    # adjacency list representation. maps interaction ids to list of target nodes, sorted in descending order by score
+    scores = {}
+
+    center_protein_dcid = escape(request.json['proteinDcid'])
```

Comment:
Oh I added this because there was a code scanning error about an [XSS vulnerability](https://owasp.org/www-community/attacks/xss/).  I think it's like if an attacker gets ahold of a victim's computer they might be able to make a request to this endpoint with proteinDcid set to some malicious script and then the script could run in the victim's browser if proteinDcid is somehow incorporated into HTML.  Should we just ignore it?

Reply:
Ok I took it out - the markup symbols might also make it harder to write tests but I'm not sure about this.